### PR TITLE
ASTNode proxy class

### PR DESCRIPTION
**Pull request by aravij**
ASTNode is a proxy class, that represent AST node and hides `networkx` API. It provides children iterator and easy field access.
Other changes:

- Now all nodes has attribute `line`, but if it wasn't in `javalang` it set to None
- Node attribute `type` is renamed to `node_type` to avoid collision with `javalang` fields.

### CODE REVIEW

**Comment by acheshkov** [comment]
@aravij,  can it be reviewed?

File: aibolit/ast_framework/_auxiliary_data.py [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+from javalang import tree
+
+from aibolit.ast_framework import ASTNodeType
+
+
+class ASTNodeReference(NamedTuple):
+    node_index: int
+
+
+javalang_to_ast_node_type: Dict[Type, ASTNodeType] = {
```
Comment (acheshkov) [inline aibolit/ast_framework/_auxiliary_data.py:34]:
```suggestion
javalang_to_ast_node_type: Dict[Type[javalang.ast.Node], ASTNodeType] = {
```

Is it possible to specify the type more precisely?

**Comment by acheshkov** [comment]
@rultor merge

**Comment by aravij** [comment]
@acheshkov  @lyriccoder 
```bash
python3 aibolit --version
Can't check aibolit version. Network is not available
aibolit 0.0.0
```
We are trying to get versions of `aibolit` through `pypi` with following code:
```python
def get_versions(pkg_name):
    url = f'https://pypi.python.org/pypi/{pkg_name}/json'
    releases = json.loads(requests.get(url, timeout=0.01).content)['releases']
    return sorted(releases, key=parse_version, reverse=True)
```

Why do we hack URL? Its format could get easily changed. Right now it redirects from `pypi.python.org` to `pypi.org`. It takes some time, so we don't make it in `0.01` of second. I will increase the timeout, but we better manage this task differently, without hacking URL.

**Comment by aravij** [comment]
@acheshkov We can try to merge it again, hope it will work.

**Comment by acheshkov** [comment]
@

> We are trying to get versions of `aibolit` through `pypi` with following code:

Good question. @lyriccoder could you answer? 
@aravij could you create an issue?

**Comment by acheshkov** [comment]
@rultor merge

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@db6acc48-2b9d-4b45-9205-875832e6af9d, thread: pool-623-thread-1
com.zerocracy.radars.github.ReSafe[81] java.io.IOException: com.zerocracy.radars.github.GhProjectNotFoundException: Github project for 'cqfn/aibolit' repo was not found
com.zerocracy.radars.github.GhProject[69] com.zerocracy.radars.github.GhProjectNotFoundException: Github project for 'cqfn/aibolit' repo was not found
```

[0.54.5](https://github.com/zerocracy/farm): Issue: cqfn/aibolit#506, Action: `created`

**Comment by aravij** [comment]
OMG, this merge is kind of freaking me out.
@acheshkov I seems like you have first reapprove the PR and then ask rultor to merge(

**Comment by acheshkov** [comment]
@rultor merge

**Pull request merged**